feat: pdf뷰어 구현 및 디렉토리 업데이트

JuhwanLee99 opened the pull request:
feat: PDF 리더 기능 초기 구현

- pdfrx 패키지를 사용한 PDF 렌더링 구현
- pdf 스크롤 가능
- 디렉토리 구성 수정 및  README.md에 반영
- 이후 텍스트 하이라이트 및 북마크 기능 추가 예정

JuhwanLee99 committed:
feat: pdf뷰어 구현 및 디렉토리 업데이트

JuhwanLee99 committed:
feat: pdf뷰어 구현 및 디렉토리 업데이트

JuhwanLee99 committed:
fix: test에서 문제가 있었던 deprecated 속성 및 const 적용

jaeho0718 reviewed (commented):
앞으로 커밋시 세부 기능별로 해주세요.
예를들어 core/theme에서 git commit -m "feat: Theme 추가"
features/snapfig 완료시 git commit -m "feat: Pdf viewer 추가"
위와 같이 해야 차후 변경이력 추적시 용이합니다~

JuhwanLee99 committed:
docs: README의 디렉토리 부분 수정

JuhwanLee99 committed:
docs: README의 디렉토리 부분 수정

JuhwanLee99 committed:
refactor: main.dart의 pdd_viewer_screen import 경로 반영

JuhwanLee99 committed:
refactor: main.dart의 pdd_viewer_screen import 경로 반영

JuhwanLee99 committed:
refactor: 이름 기반 라우팅(navigation)을 위해 initialRoute + route 추가

JuhwanLee99 committed:
refactor: 이름 기반 라우팅(navigation)을 위해 initialRoute + route 추가

JuhwanLee99 committed:
refactor: 내부 캐싱과 에러 핸들링을 더해 싱글톤 패턴으로 리팩터링

JuhwanLee99 committed:
refactor: 내부 캐싱과 에러 핸들링을 더해 싱글톤 패턴으로 리팩터링

JuhwanLee99 committed:
refactor: FutureBuilder를 이용해 PdfService로부터 PDF를 비동기 로드하고, 로딩 중·오류·완료 상태를 처리하도록 수정. 렌더링은 위젯으로 이동.

JuhwanLee99 committed:
refactor: FutureBuilder를 이용해 PdfService로부터 PDF를 비동기 로드하고, 로딩 중·오류·완료 상태를 처리하도록 수정. 렌더링은 위젯으로 이동.

JuhwanLee99 committed:
refactor: pdf 렌더링을 위젯으로 처리하기 위함

JuhwanLee99 committed:
refactor: pdf 렌더링을 위젯으로 처리하기 위함

JuhwanLee99 committed:
style: 주석수정

JuhwanLee99 committed:
style: 주석수정

JuhwanLee99 committed:
refactor: 디렉토리 변경

JuhwanLee99 committed:
refactor: 디렉토리 변경

JuhwanLee99 committed:
refactor: 디렉토리 변경

JuhwanLee99 committed:
Delete lib/features/snapfig directory

JuhwanLee99 committed:
Delete lib/features/snapfig directory

JuhwanLee99 committed:
refactor: 내부 package import 방식 변경

JuhwanLee99 committed:
refactor: 내부 package import 방식 변경

JuhwanLee99 committed:
fix: 실행시 pubspec.yaml의 name을 따르도록 수정해 빌드 오류 수정

JuhwanLee99 committed:
Merge remote-tracking branch 'origin/feature/Pdf_Reader' into feature/Pdf_Reader

JuhwanLee99 committed:
fix: 실행시 pubspec.yaml의 name을 따르도록 수정해 빌드 오류 수정

JuhwanLee99 committed:
fix: 실행시 pubspec.yaml의 name을 따르도록 수정해 빌드 오류 수정

JuhwanLee99 committed:
fix: 앱 실행시 첫 페이지가 배경으로 나오는 오류 수정

JuhwanLee99 committed:
fix: 앱 실행시 첫 페이지가 배경으로 나오는 오류 수정

JuhwanLee99 committed:
feat: 핀치 제스처로 확대 축소 기능 구현

JuhwanLee99 committed:
feat: 핀치 제스처로 확대 축소 기능 구현

JuhwanLee99 committed:
refactor: 코드를 StatefulWidget 형식으로 개선

JuhwanLee99 committed:
refactor: 코드를 StatefulWidget 형식으로 개선

JuhwanLee99 committed:
chore: vector_math 패키지 추가

JuhwanLee99 committed:
chore: vector_math 패키지 추가

JuhwanLee99 committed:
feat: 아이콘으로 확대,축소,중앙정렬 기능 구현

JuhwanLee99 committed:
feat: 아이콘으로 확대,축소,중앙정렬 기능 구현

JuhwanLee99 committed:
feat: 아이콘으로 화면 폭/높이 맞춤 기능 추가

JuhwanLee99 committed:
feat: 아이콘으로 화면 폭/높이 맞춤 기능 추가

JuhwanLee99 committed:
style: 주석 추가

JuhwanLee99 committed:
style: 주석 추가

JuhwanLee99 committed:
feat: 로컬 저장소에서 파일 불러오기 기능 추가

JuhwanLee99 committed:
feat: 로컬 저장소에서 파일 불러오기 기능 추가

JuhwanLee99 committed:
chore: shared_preferences 패키지 추가

JuhwanLee99 committed:
chore: shared_preferences 패키지 추가

JuhwanLee99 committed:
feat: 최근파일 목록 기능 추가

JuhwanLee99 committed:
feat: 최근파일 목록 기능 추가

jaeho0718 committed:
feat: pdf_core에 모델 추가

- pubspec.yaml에 isar, isar_flutter_libs, path_provider 추가
- pubspec.lock에 새로운 패키지 버전 및 의존성 추가
- BasePDF, BasePage, BaseLayout 인터페이스 추가

jaeho0718 committed:
feat: pdf_core에 모델 추가

- pubspec.yaml에 isar, isar_flutter_libs, path_provider 추가
- pubspec.lock에 새로운 패키지 버전 및 의존성 추가
- BasePDF, BasePage, BaseLayout 인터페이스 추가

JuhwanLee99 committed:
chore: cupertino, isar 패키지 추가

JuhwanLee99 committed:
chore: cupertino, isar 패키지 추가

JuhwanLee99 committed:
docs: README 업데이트(체리픽 반영)

JuhwanLee99 committed:
docs: README 업데이트(체리픽 반영)

JuhwanLee99 committed:
chore: ocr_result sample 추가

JuhwanLee99 committed:
chore: ocr_result sample 추가

JuhwanLee99 committed:
chore: path 패키지 추가

JuhwanLee99 committed:
chore: path 패키지 추가

JuhwanLee99 committed:
feat: page와 figure 사이드바 구현

JuhwanLee99 committed:
feat: page와 figure 사이드바 구현

JuhwanLee99 committed:
chore: figure 테스트용 pdf 파일 추가

JuhwanLee99 committed:
chore: figure 테스트용 pdf 파일 추가

JuhwanLee99 committed:
chore: figure 테스트용 pdf 파일 교체

JuhwanLee99 committed:
chore: figure 테스트용 pdf 파일 교체

JuhwanLee99 committed:
fix: json파일의 데이터가 figure 목록에 보이도록 수정

JuhwanLee99 committed:
fix: json파일의 데이터가 figure 목록에 보이도록 수정

JuhwanLee99 committed:
feat: pdf뷰어에 하단바(확대율, 페이지번호, 검색) 기능 추가

JuhwanLee99 committed:
feat: pdf뷰어에 하단바(확대율, 페이지번호, 검색) 기능 추가

JuhwanLee99 committed:
fix: 실제 확대율이 표시되도록 수정

JuhwanLee99 committed:
fix: 실제 확대율이 표시되도록 수정

JuhwanLee99 committed:
fix: 디렉토리에서 pdf 선택시 새로고침 되지 않는 버그 수정

JuhwanLee99 committed:
fix: 디렉토리에서 pdf 선택시 새로고침 되지 않는 버그 수정

JuhwanLee99 committed:
feat: pdf 스크롤시 페이지 목록의 중앙을 현재 페이지로 정렬

JuhwanLee99 committed:
feat: pdf 스크롤시 페이지 목록의 중앙을 현재 페이지로 정렬

JuhwanLee99 committed:
fix: 핀치 제스처로 비율 변경시에도 수치가 반영되도록 수정

JuhwanLee99 committed:
fix: 핀치 제스처로 비율 변경시에도 수치가 반영되도록 수정

JuhwanLee99 committed:
refactor: pdf파일을 path 인자와 isAsset 여부로 로드

JuhwanLee99 committed:
refactor: pdf파일을 path 인자와 isAsset 여부로 로드

JuhwanLee99 committed:
feat: 홈화면(뒤로가기)이동 아이콘 배치, 사이드바 숨기기 아이콘 위치 이동

JuhwanLee99 committed:
feat: 홈화면(뒤로가기)이동 아이콘 배치, 사이드바 숨기기 아이콘 위치 이동

jaeho0718 committed:
Merge branch 'feature/Pdf_Reader' of https://github.com/CommitBang/Frontend into feature/Pdf_Reader

jaeho0718 committed:
fix: package명 snapfig로 변경

JuhwanLee99 committed:
fix: package 이름을 snapfig로 변경

JuhwanLee99 committed:
fix: 중복된 dependencies 정리

JuhwanLee99 committed:
fix: import 주소 수정

JuhwanLee99 committed:
refactor: pdf_viewer에서 사용되는 model 구현체 디렉토리 이동

JuhwanLee99 committed:
docs: 코드 refactor 반영 README 업데이트

JuhwanLee99 committed:
fix: super.key 반영

JuhwanLee99 committed:
feat: 홈화면에서 열기를 누르면 뷰어로 이동하는 기능 추가

JuhwanLee99 committed:
Revert "fix: super.key 반영"

This reverts commit bc145bc3d6f5fdbdd81c9af00c945861d435f426.

JuhwanLee99 committed:
Revert "feat: 홈화면에서 열기를 누르면 뷰어로 이동하는 기능 추가"

This reverts commit b76fed8fc9875b999e59b66d709686822f26cf2e.

JuhwanLee99 committed:
fix: pdf뷰어 아이콘 동작 수정
- 사이드바 숨기기 아이콘 이동
- 홈화면으로 뒤로가기 아이콘
